crates/ibc/src/core/ics03_connection/handler/conn_open_init.rs

```diff
@@ -20,9 +19,7 @@ where
     client_state_of_b_on_a.confirm_not_frozen()?;
 
     if let Some(version) = msg.version {
-        if !ctx_a.get_compatible_versions().contains(&version) {
-            return Err(ConnectionError::VersionNotSupported { version }.into());
-        }
+        version.verify_version_supported(&ctx_a.get_compatible_versions())?;
```

Comment:
I find what we had before more readable. I read both as:

+ Before: ensure that the message's version is in the list of "compatible versions"
+ After: ensure that the message's version is supported, based on the list of compatible versions.
    + I'm not sure just by reading this what the requirement for "being supported" is

Reply:
Ah, it seems like we were not doing the right check (looking at `verify_version_supported()` and ibc-go's `IsSupportedVersion`